src/main/java/org/jenkinsci/plugins/oic/OicServerManualConfiguration.java

```diff
@@ -156,6 +156,11 @@ public OIDCProviderMetadata toProviderMetadata() {
             ArrayList<JWSAlgorithm> allAlgorthms = new ArrayList<>();
             allAlgorthms.addAll(JWSAlgorithm.Family.HMAC_SHA);
             allAlgorthms.addAll(JWSAlgorithm.Family.SIGNATURE);
+
+            if (!allAlgorthms.isEmpty() && OicAlgorithmValidator.isJwsAlgoFipsNonCompliant(allAlgorthms)) {
+                throw new IllegalArgumentException(
```

Comment:
Are we sure of this?
This will always launch an exception if there's a non compliant algorithm in the nimbus library (I guess there wil lalways be), and the user can not do anything to avoid it.
We should instead filter the list to allow only compliant algorithms